src/analyses/ai.cpp

```diff
@@ -387,7 +398,16 @@ bool ai_baset::do_function_call_rec(
   const goto_functionst &goto_functions,
   const namespacet &ns)
 {
-  assert(!goto_functions.function_map.empty());
+  PRECONDITION(!goto_functions.function_map.empty());
+
+  // We can't really do this here -- we rely on
+  // these being removed by some previous analysis.
+  PRECONDITION(function.id()!=ID_dereference);
```

Comment:
It looks like previously this function would still return if it got a dereference, where as now it's going to hit a precondition... Is that a change in behaviour? And does this somehow prevent us ever using AI to resolve function pointers?

Reply:
Previously it would ignore function pointers, giving unsound results.  However it was always used with the implicit (undocumented!) precondition that function pointers and returns had been removed.

It doesn't prevent us handling function pointers; we'd just need to add the branch back in, once we'd dealt with the return removal issues.